Toggl.Foundation.MvvmCross/ViewModels/SignupViewModel.cs

```diff
@@ -305,14 +305,27 @@ private void onCompleted()
             signupDisposable = null;
         }
 
-        public void GoogleSignup()
+        public async Task GoogleSignup()
         {
+            if (!countryId.HasValue)
+            {
+                shakeSubject.OnNext(ShakeTargets.Country);
+                return;
+            }
+
+            if (!termsOfServiceAccepted)
+                termsOfServiceAccepted = await navigationService.Navigate<bool>(typeof(TermsOfServiceViewModel));
+
+            if (!termsOfServiceAccepted)
+                return;
+
             if (isLoadingSubject.Value) return;
 
             isLoadingSubject.OnNext(true);
+            errorMessageSubject.OnNext("");
```

Comment:
I don't really like `""` for empty strings, consider `string.Empty`:

```suggestion
            errorMessageSubject.OnNext(string.Empty);
```